test/e2e/aws_test.go

```diff
@@ -118,6 +140,172 @@ var _ = Describe("functional tests", func() {
 	})
 })
 
+func scaleMachineDeployment(namespace, machineDeployment string, replicasCurrent int32, replicasProposed int32) {
```

Comment:
You shouldn't need `replicasCurrent` here, you should be able to just use the value returned from getting the MachineDeployment instead.

Reply:
From testing perspective, i want to ensure that the MachineDeployment is getting updated properly. Hence, not depending on MachineDeployment.